src/cmd/builtin/shutdown.c

```diff
@@ -12,13 +12,54 @@
 # include <config.h>
 #endif
 #include <unistd.h>
+#include <jansson.h>
 #include <flux/core.h>
 
 #include "src/broker/state_machine.h"
+#include "src/common/libkvs/kvs_checkpoint.h"
 #include "src/common/libutil/uri.h"
 
 #include "builtin.h"
 
+static void get_checkpoint_sequence (flux_t *h, int *seq) {
+    flux_future_t *f;
+    (*seq) = 0;
+    if (!(f = kvs_checkpoint_lookup (h, NULL, 0))
+        || (kvs_checkpoint_lookup_get_sequence (f, seq) < 0
+            && errno != ENOENT))
+        log_msg_exit ("Error fetching checkpoint sequence: %s",
+                      future_strerror (f, errno));
+}
```

Comment:
Quick comment as I head out to a doctor's apt:
- open brace for the function should be on a line by itself per project style
- I think we'll want to check the current live kvs root sequence number, not the stored checkpoint since the latter will be out of date